src/Admin/SiteHealth.php

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Class SiteHealth.
+ *
+ * @package AMP
+ */
+
+namespace Amp\AmpWP\Admin;
+
+use AMP_Options_Manager;
+use AMP_Theme_Support;
+use AMP_Post_Type_Support;
+
+/**
+ * Class SiteHealth
+ *
+ * Adds tests and debugging information for Site Health.
+ *
+ * @since 1.5.0
+ */
+class SiteHealth {
+
+	/**
+	 * Adds the filters.
+	 */
+	public function init() {
+		add_filter( 'site_status_tests', [ $this, 'add_tests' ] );
+		add_filter( 'debug_information', [ $this, 'add_debug_information' ] );
+		add_filter( 'site_status_test_php_modules', [ $this, 'add_extensions' ] );
+	}
+
+	/**
+	 * Adds Site Health tests related to this plugin.
+	 *
+	 * @param array $tests The Site Health tests.
+	 * @return array $tests The filtered tests.
+	 */
+	public function add_tests( $tests ) {
+		$direct_tests = [
+			'persistent_object_cache' => esc_html__( 'Persistent object cache', 'amp' ),
+			/* translators: %s: a type of PHP function */
+			'curl_multi_functions'    => sprintf( esc_html__( '%s functions', 'amp' ), 'curl_multi_*' ),
+			'icu_version'             => esc_html__( 'ICU version', 'amp' ),
+		];
+
+		foreach ( $direct_tests as $test_name => $test_label ) {
+			$tests['direct'][ $test_name ] = [
+				'label' => $test_label,
+				'test'  => [ $this, $test_name ],
+			];
+		}
+
+		return $tests;
+	}
+
+	/**
+	 * Gets the test result data for whether there is a persistent object cache.
+	 *
+	 * @return array The test data.
+	 */
+	public function persistent_object_cache() {
+		$is_using_object_cache = wp_using_ext_object_cache();
+		$data                  = [
+			'badge'       => [
+				'label' => esc_html__( 'AMP', 'amp' ),
+				'color' => $is_using_object_cache ? 'green' : 'orange',
+			],
+			'description' => esc_html__( 'The AMP plugin performs at its best when persistent object cache is enabled.', 'amp' ),
+			'actions'     => '',
+			'test'        => 'persistent_object_cache',
```

Comment:
Let's prefix all tests with `amp_`

Reply:
Good idea, https://github.com/ampproject/amp-wp/pull/3886/commits/60970cebf3586cac5e6aaae23311ce66e8e7f5fb does that.